src/lib/storage/fixed_size_attribute_vector.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vector>
+#include "storage/base_attribute_vector.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+template <typename T>
+class FixedSizeAttributeVector : public BaseAttributeVector {
+ public:
+  explicit FixedSizeAttributeVector(std::vector<T> value_ids) : _value_ids{value_ids} {}
+
+  // returns the value id at a given position
+  virtual ValueID get(const size_t index) const {
+    DebugAssert(index < size(), "Index out of bounds");
+    return ValueID(_value_ids[index]);
+  }
+
+  // sets the value id at a given position
+  virtual void set(const size_t i, const ValueID value_id) {
+    Assert(i < _value_ids.size(), "Out of range");
```

Comment:
If you decide to always assert this, you could just use the `at()` method, right?